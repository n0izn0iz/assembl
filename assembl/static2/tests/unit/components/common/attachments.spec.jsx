import React from 'react';
import renderer from 'react-test-renderer';

import Attachments from '../../../../js/app/components/common/attachments';

describe('Attachments component', () => {
  it('should render the list of attachments', () => {
    const attachments = [
      {
        id: 'foo',
        title: 'Foo',
        externalUrl: 'http://www.example.com/foo'
      },
      {
        id: 'bar',
        title: 'Bar.jpg',
        externalUrl: 'http://www.example.com/bar.jpg',
        mimeType: 'image/jpeg'
      }
    ];
    const component = renderer.create(<Attachments attachments={attachments} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});