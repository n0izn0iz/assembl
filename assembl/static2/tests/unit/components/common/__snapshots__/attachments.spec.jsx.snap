// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attachments component should render the list of attachments 1`] = `
<div
  className="attachments"
>
  <div
    className="attachment"
  >
    <span
      className="assembl-icon-synthesis"
    />
    <span
      className="title"
    >
      Foo
    </span>
    <a
      download="Foo"
      href="http://www.example.com/foo.pdf"
      rel="noopener noreferrer"
      target="_blank"
      type="application/pdf"
    >
      <span
        className={undefined}
        style={undefined}
      >
        Download
      </span>
    </a>
  </div>
  <div
    className="attachment"
  >
    <span
      className="assembl-icon-synthesis"
    />
    <span
      className="title"
    >
      Bar.jpg
    </span>
    <a
      download="Bar.jpg"
      href="http://www.example.com/bar.jpg"
      rel="noopener noreferrer"
      target="_blank"
      type="image/jpeg"
    >
      <span
        className={undefined}
        style={undefined}
      >
        Download
      </span>
    </a>
  </div>
</div>
`;
